### PR TITLE
refactor(metrics): unit tests for Metrics are written to follow similar convention as Logger/Tracer

### PR DESCRIPTION
**Pull request by arnabrahman**
## Description of your changes
Metrics unit tests were not following the same convention as Logger/Tracer package. In Logger/Tracer tests, only public functions are unit tested. But for Metrics, unit tests are written per feature/behavior. 

This PR updates the tests to match the same convention as Logger/Tracer package. Only public methods are tested, following the recommendations here: [Best Practices](https://github.com/goldbergyoni/javascript-testing-best-practices/#-%EF%B8%8F-14-stick-to-black-box-testing-test-only-public-methods)

100% test coverage is maintained. Some additional housekeeping is implemented, `constants` & `helpers` are introduced for better code maintainability.

I've removed the previous feature tests because in my understanding leaving them will create further confusion about the testing patterns. I can add them again if maintainers feel the current sets of tests in this PR are inadequate. 

### How to verify this change
All the tests will run successfully along with 100% test coverage.

### Related issues, RFCs

**Issue number:** #163 

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-typescript/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [ ] I have made corresponding changes to the *documentation*
- [ ] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [x] Any *dependent changes have been merged and published*
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

***Is it a breaking change?:*** NO

- [ ] I have documented the migration process
- [ ] I have added, implemented necessary warnings (if it can live side by side)

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by dreamorosi** [comment]
Hi @arnabrahman thank you so much for opening this PR!

Given the size, please allow us some time to go through it and review it.

I'll get back to you here with comments as I review it.

File: packages/metrics/src/helpers.ts
Comment (dreamorosi) [inline packages/metrics/src/helpers.ts:1]: We are going to deprecate/remove these helpers in the next major release from all other utilities (i.e. `createLogger`) so I think we could avoid adding it here and use `new Metrics(...)` directly in the tests.

File: packages/metrics/tests/helpers/metricsUtils.ts [hunk 1]
```diff
@@ -21,4 +25,27 @@ const getMetrics = async (cloudWatchClient: CloudWatch, namespace: string, metri
   }, retryOptions);
 };
 
-export { getMetrics };
\ No newline at end of file
+const setupDecoratorLambdaHandler = (metrics: Metrics, options: ExtraOptions = {}): Handler => {
```
Comment (dreamorosi) [inline packages/metrics/tests/helpers/metricsUtils.ts:28]: Very nice

File: packages/metrics/src/helpers.ts
Comment (arnabrahman) [reply packages/metrics/src/helpers.ts:1]: If this is the case then dropping `createMetrics` helper function will be logical. I will replace it then.

Comment (dreamorosi) [reply packages/metrics/src/helpers.ts:1]: Thanks, that'd be great

**Review by dreamorosi — COMMENTED** [review]
I've done a first review and things are looking good!

I have left one comment about the helper function, that I think we could drop.

Now I'm going to focus on reviewing the actual test cases and see if all the previous ones are covered.

File: packages/metrics/src/helpers.ts
Comment (arnabrahman) [reply packages/metrics/src/helpers.ts:1]: I've addressed this here [af2a6b8](https://github.com/awslabs/aws-lambda-powertools-typescript/pull/1414/commits/af2a6b8388a4f2284e5549b04dfe9b63c8d48f1a), [a063546](https://github.com/awslabs/aws-lambda-powertools-typescript/pull/1414/commits/a063546d37bf58013574705a6ec701cdc387dc23), [4037c3c](https://github.com/awslabs/aws-lambda-powertools-typescript/pull/1414/commits/4037c3c10d9793c76b867e6d9bcf0091b40b031e)

**Review by dreamorosi — APPROVED** [review]
Hi @arnabrahman thank you so much for the work you've done here.

I really appreciate the effort and dedication you've put into improving the project.

**Comment by arnabrahman** [comment]
Happy to help

**Pull request merged**